gimie/sources/common/license.py

```diff
@@ -1,56 +1,125 @@
+    query = """
+       query {
+         repository(owner: "%s", name: "%s") {
+           defaultBranchRef {
+             name
+           }
+           object(expression: "HEAD:") {
+             ... on Tree {
+               entries {
+                 name
+               }
+             }
+           }
+         }
+       }
+       """ % (
+        username,
+        repo_name,
+    )
+    # Send a GET request to the GitHub API to get repository information
+    response = requests.post(url, json={"query": query}, headers=headers)
 
-    Examples
-    --------
-    >>> locate_licenses('.')
-    ['./LICENSE']
-    """
+    # Check if the request was successful (status code 200)
+    if response.status_code == 200:
+        repository_info = response.json()
+        default_branch_name = repository_info["data"]["repository"][
+            "defaultBranchRef"
+        ]["name"]
+        files_dict = repository_info["data"]["repository"]["object"]["entries"]
+        try:
+            return default_branch_name, files_dict
+        except KeyError:
+            print("Could not identify default branch")
+
+
+def get_license_path(repo_url, files_dict) -> str:
+    """Given a list of files, returns the URL filepath which contains the license"""
+    repo_url = repo_url.rstrip("/")
     license_files = []
-    pattern = r".*(license(s)?|reus(e|ing)|copy(ing)?)(\.(txt|md|rst))?$"
-
-    for root, _, files in os.walk(path):
-        # skip toplevel hidden dirs (e.g. .git/)
-        subdir = os.path.relpath(root, path)
-        if subdir.startswith(".") and subdir != ".":
-            continue
-        for file in files:
-            # skip hidden files
-            if file.startswith("."):
+    if files_dict:
+        for file in files_dict:
+            if file["name"].startswith("."):
                 continue
-
-            if re.match(pattern, file, flags=re.IGNORECASE):
-                license_path = os.path.join(root, file)
+            pattern = (
+                r".*(license(s)?|reus(e|ing)|copy(ing)?)(\.(txt|md|rst))?$"
+            )
+            if re.match(pattern, file["name"], flags=re.IGNORECASE):
+                license_path = (
+                    repo_url
+                    + f"/blob/{get_default_branch_name_and_root_files_dict(repo_url)[0]}/"
+                    + file["name"]
+                )
                 license_files.append(license_path)
+                print(
+                    license_path
+                    + " is the license path gimie found, starting extraction of license..."
+                )
+
+    if len(license_files) > 1:
+        return "More than 1 license file was found, please make sure you only have one license."
+    else:
+        for license_file_url in license_files:
+            return license_file_url
+
 
-        # The first root of os.walk is the current dir
-        if not recurse:
-            return license_files
+def github_read_file(url: str, github_token: str = None) -> dict:
```

Comment:
If above is implemented,this function will not be needed here (defined as a method of the file object)

Reply:
headers will then be passed down from the Extractor to the File